src/screens/Search/Search.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from "react";
+import { View, Image, FlatList } from "react-native";
+import { Typography } from "../../components/Typography";
+import { useSearch } from "../../context/search";
+import { ScreenType } from "../../navigation/types";
+import { styles } from "./styles";
+import { Product } from "../../components/ProductSlide";
+import interrogrationMark from "../../../assets/images/interrogationMark.png";
+import data from "./data.json";
+import SearchProducts from "../../components/SearchProducts/SearchProducts";
+import { api } from "../../service";
+
+type Props = ScreenType<"Search">;
+
+export const Search: React.FC<Props> = () => {
+
+  const [products, setProducts] = useState<Product[]>([]);
+  const [search, setSearch] = useSearch();
+  const [data, setData] = useState<Product[]>([]);
+
+  useEffect(() => {
+    api.get("/products").then((response) => {
+      const { data } = response;
+      setProducts(data);
+      setData(data);
+    });
+  }, []);
```

Comment:
It better use async function here, like dashboard screen. 

Also you can use params to seach:
```suggestion
  useEffect(() => {
    api.get("/products", {
          params: {
            filter: {
              search,
            },
          },
        }).then((response) => {
      const { data } = response;
      setProducts(data);
      setData(data);
    });
  }, []);
```